lib/services/user.go

```diff
@@ -491,8 +491,8 @@ func (u *UserV1) Check() error {
 		return trace.BadParameter("user name cannot be empty")
 	}
 	for _, login := range u.AllowedLogins {
-		_, err := parse.RoleVariable(login)
-		if err == nil {
+		e, err := parse.NewExpression(login)
```

Comment:
New logic only returns error in case of grammatic errors and accepts literal values. Doesn't it mean that now `err` should return error here if it's not nil? Otherwise the following will be accepted: `{{regexp_not_match(...)` which probably should not be ignored as it's a typo and we should let users know